fastembed/image/transform/functional.py

```
@@ -145,3 +145,77 @@ def pad2square(
     new_image = Image.new(mode="RGB", size=(size, size), color=fill_color)
     new_image.paste(image.crop((left, top, right, bottom)) if crop_required else image)
     return new_image
+
+
+def resize_longest_edge(
+    image: Image.Image,
+    max_size: int,
+    resample: Union[int, Image.Resampling] = Image.Resampling.LANCZOS,
```

Ruff (0.14.10)

Undefined name `Union`

(F821)